skimage/transform/_polar.pyx

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+from libc.math cimport sqrt, sin, cos, floor, ceil, M_PI
+from skimage._shared.interpolation cimport round
+
+
+def cart2pol(img, center=None, full_output=False):
+    """
+    Polar transform of an image.
+
+    Parameters
+    ----------
+
+    img : (M, N, D) ndarray
+        Input image.
+    center : (row, column) tuple or (2,) ndarray, optional
+        Center of the polar transform.
+    full_output : boolean, optional (default False)
+        Extend the output to fully enclose the image.
+    """
+    cdef:
+        int deltax, delta_y, radius
+        int center0, center1, corner0, corner1
+        Py_ssize_t imgshape0 = img.shape[0]
+        Py_ssize_t imgshape1 = img.shape[1]
+        double sina, cosa, a, r
+        int angle = 360
+        int i, j
+        double pi_over_180 = M_PI / 180.
+
+    # Default center.
+    if not center:
+        center0 = <int> round(imgshape0/2.)
+        center1 = <int> round(imgshape1/2.)
+    else:
+        center0 = <int> center[0]
+        center1 = <int> center[1]
+
+    # Calculate the radius
+    if full_output:
```

Comment:
This is a very big block. I would break it out into another function if possible.